spec/features/idv/steps/in_person/state_id_50_50_spec.rb

```diff
@@ -0,0 +1,378 @@
+          with: InPersonHelper::GOOD_IDENTITY_DOC_CITY,
+        )
+        expect(page).to have_field(
+          t('in_person_proofing.form.state_id.zipcode'),
+          with: InPersonHelper::GOOD_IDENTITY_DOC_ZIPCODE,
+        )
+        expect(page).to have_field(
+          t('in_person_proofing.form.state_id.identity_doc_address_state'),
+          with: Idp::Constants::MOCK_IDV_APPLICANT[:state_id_jurisdiction],
+        )
+        expect(page).to have_checked_field(
+          t('in_person_proofing.form.state_id.same_address_as_id_yes'),
+          visible: false,
+        )
+      end
+    end
+  end
+
+  context 'when updating state id info from the verify info page' do
+    let(:first_name_update) { 'Natalya' }
+
+    context 'when the controller is switched from enabled to disabled' do
+      before do
+        allow(IdentityConfig.store).to receive(
+          :in_person_state_id_controller_enabled,
+        ).and_return(true)
+        visit_idp_from_sp_with_ial2(:oidc, **{ client_id: ipp_service_provider.issuer })
+        sign_in_via_branded_page(user)
+        begin_in_person_proofing_with_opt_in_ipp_enabled_and_opting_in
+        complete_prepare_step(user)
+        complete_location_step
+        complete_state_id_controller(user, same_address_as_id: true)
+        complete_ssn_step(user)
+        expect_in_person_step_indicator_current_step(t('step_indicator.flows.idv.verify_info'))
+        allow(IdentityConfig.store).to receive(
+          :in_person_state_id_controller_enabled,
+        ).and_return(false)
+        click_button t('idv.buttons.change_state_id_label')
+        fill_in t('in_person_proofing.form.state_id.first_name'), with: first_name_update
+        click_button t('forms.buttons.submit.update')
+      end
+
+      it 'navigates back to the verify_info page' do
+        expect(page).to have_current_path(idv_in_person_verify_info_path)
+        expect(page).to have_text(first_name_update)
+      end
+    end
+
+    context 'when the controller is switched from disabled to enabled' do
+      before do
+        allow(IdentityConfig.store).to receive(
+          :in_person_state_id_controller_enabled,
+        ).and_return(false)
+        visit_idp_from_sp_with_ial2(:oidc, **{ client_id: ipp_service_provider.issuer })
+        sign_in_via_branded_page(user)
+        begin_in_person_proofing_with_opt_in_ipp_enabled_and_opting_in
+        complete_prepare_step(user)
+        complete_location_step
+        complete_state_id_step(user, same_address_as_id: true)
+        complete_ssn_step(user)
+        expect_in_person_step_indicator_current_step(t('step_indicator.flows.idv.verify_info'))
+        allow(IdentityConfig.store).to receive(
+          :in_person_state_id_controller_enabled,
+        ).and_return(true)
+        click_button t('idv.buttons.change_state_id_label')
+        fill_in t('in_person_proofing.form.state_id.first_name'), with: first_name_update
```

Comment:
Do you think we should check path here to show controller state id form?

Reply:
The tests starting at line 105 are covering this scenario.